clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/AutoElasticsearchIndex.scala

```diff
@@ -14,59 +15,106 @@ import scala.reflect.runtime.universe.Type
 ) extends ElasticsearchIndex[Document] {
+  override val indexName: String =
+    if (version == 1) {
+      // We started out with no version suffix, so keep it that way for v1.
+      name
+    } else {
+      s"${name}_v$version"
+    }
+
   override def fields: Seq[FieldDefinition] =
-    AutoElasticsearchIndex.getFieldDefinitions[Document]
+    AutoElasticsearchIndex.getFieldDefinitions[Document](version)
 }
 
 object AutoElasticsearchIndex {
-  private def getFieldDefinitions[Document: FieldMapper]
-    : Seq[FieldDefinition] = {
+  private def getFieldDefinitions[Document: FieldMapper](
+    version: Int
+  ): Seq[FieldDefinition] = {
+    val stringToDef = stringToDefinition(version)
+
     FieldMapper[Document].fields.toSeq.sortBy {
       case (name, _) => name
     } map {
-      case (name, value) => getFieldDefinition(name, value)
+      case (name, value) =>
+        stringToDef(value)(ElasticsearchUtil.toElasticsearchName(name))
     }
   }
 
-  private def getFieldDefinition(fieldName: String,
-                                 fieldType: Type): FieldDefinition = {
+  private def stringToDefinitionV1(
+    fieldType: Type
+  ): (String => FieldDefinition) = {
     import scala.reflect.runtime.universe.typeOf
-    val stringToDefinition = fieldType match {
-      case tpe if tpe =:= typeOf[Boolean]           => booleanField _
-      case tpe if tpe =:= typeOf[Int]               => intField _
-      case tpe if tpe =:= typeOf[Long]              => longField _
-      case tpe if tpe =:= typeOf[Float]             => floatField _
-      case tpe if tpe =:= typeOf[Double]            => doubleField _
-      case tpe if tpe =:= typeOf[String]            => keywordField _
-      case tpe if tpe <:< typeOf[EnumEntry]         => keywordField _
-      case tpe if tpe =:= typeOf[OffsetDateTime]    => dateField _
-      case tpe if tpe =:= typeOf[UUID]              => keywordField _
-      case tpe if tpe =:= typeOf[UpsertId]          => keywordField _
-      case tpe if tpe =:= typeOf[Option[Boolean]]   => booleanField _
-      case tpe if tpe =:= typeOf[Option[Int]]       => intField _
-      case tpe if tpe =:= typeOf[Option[Long]]      => longField _
-      case tpe if tpe =:= typeOf[Option[Float]]     => floatField _
-      case tpe if tpe =:= typeOf[Option[Double]]    => doubleField _
-      case tpe if tpe =:= typeOf[Option[String]]    => keywordField _
-      case tpe if tpe <:< typeOf[Option[EnumEntry]] => keywordField _
+    fieldType match {
+      case tpe if tpe =:= typeOf[Boolean]           => booleanField
+      case tpe if tpe =:= typeOf[Int]               => intField
+      case tpe if tpe =:= typeOf[Long]              => longField
+      case tpe if tpe =:= typeOf[Float]             => floatField
+      case tpe if tpe =:= typeOf[Double]            => doubleField
+      case tpe if tpe =:= typeOf[String]            => keywordField
+      case tpe if tpe =:= typeOf[Symbol]            => keywordField
```

Comment:
I chose `Symbol` to represent arbitrary things that should map to keywords because I didn't see the point in making another class that would do nothing but wrap arbitrary `String`s (as opposed to `UpsertId`, which also contains unique generation and parsing logic).